jujugui/static/gui/src/app/components/deployment/deployment.js

```diff
@@ -31,6 +31,39 @@ YUI.add('deployment-component', function() {
       getUnplacedUnitCount: React.PropTypes.func.isRequired,
     },
 
+    componentDidMount: function() {
```

Comment:
If you don't navigate to a particular step in the flow this check allows us to start the flow from wherever a user should start. e.g. if a user doesn't already have a jaas account we'll want them to start at the signup step, if they have already signed up we'll want them to start at choosing a cloud, if they've chosen a cloud we'll want to start them at the deploy summary.

See my comments here: https://github.com/juju/juju-gui/pull/1485/files#diff-760cf384f3bb1b0f33ff2b63af7534f9R43

The check for only changing state if there is no active component provided is here: https://github.com/juju/juju-gui/pull/1485/files#diff-760cf384f3bb1b0f33ff2b63af7534f9R53

The default until we have signup and a way to check if we've deployed is here: https://github.com/juju/juju-gui/pull/1485/files#diff-760cf384f3bb1b0f33ff2b63af7534f9R55